### ISSUE TITLE
bookshelf, remove / delete publication does not work

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#709]
```
[2]   readium-desktop:sync ### action type API_REQUEST +37s
[2]   readium-desktop:main#redux/sagas/api publication delete e16dc6ed-1a7e-4695-a521-3f70a6f50e17 +37s
[2]   readium-desktop:main#redux/sagas/api NotFoundError { message: 'document not found' } +0ms
[2]   readium-desktop:sync ### action type API_ERROR +1ms
[2]   readium-desktop:sync ### action type API_REQUEST +4ms
[2]   readium-desktop:main#redux/sagas/api catalog get undefined +4ms
[2]   readium-desktop:sync ### action type API_REQUEST +1ms
[2]   readium-desktop:main#redux/sagas/api publication getAllTags undefined +2ms
[2]   readium-desktop:sync ### action type API_SUCCESS +14ms
[2]   readium-desktop:sync ### action type API_SUCCESS +5ms
```

**Comment by danielweck** [comment]
This is a regression bug caused by Pull Request https://github.com/readium/readium-desktop/pull/659

I tested the commit immediately before in the history of the `develop` branch, and the "delete" feature works:
https://github.com/readium/readium-desktop/commits/develop

**Comment by danielweck** [comment]
@panaC I guess the culprit is this API refactoring:

https://github.com/readium/readium-desktop/pull/659/files#diff-968212bd5d4ddfe1c6bab01fa6bf4e16L46-L49

Could you please look into this regression bug? (high priority)
Thanks :)

**Comment by danielweck** [comment]
Okay, I think I found the culprit. The spread operator on string literals "reduces" the string to the first character.
https://github.com/readium/readium-desktop/pull/659/files#diff-6d19fb16e43a410967ed30ca67675aeeR38

**Comment by danielweck** [comment]
Given the severity of the bug in the central API `processRequest()` function, I am hot-fixing this right now:

https://github.com/readium/readium-desktop/blob/926000f851597e4cbe6cd3b51a8748ce585b7790/src/main/redux/sagas/api.ts#L27-L46

@panaC feel free to create a new PR to fix this in a more elegant / more thorough manner. My fix is just a naive test for the string type, to make sure that the spread operator is not applied.

**Comment by danielweck** [comment]
Hotfix: https://github.com/readium/readium-desktop/commit/baa17321404985e5d721f8894203a745091df73b
(please improve the fix if deemed necessary, I just wanted to fix the broken develop branch as quickly as possible)

**Comment by panaC** [comment]
It's normal, i haven't modified withApi because i replace it in the next PR.
But i forgotten to updated the spread Operator in API.tsx.

It works in the next typing PR.

fix with #710

**Comment by danielweck** [comment]
PR fix: https://github.com/readium/readium-desktop/pull/710
